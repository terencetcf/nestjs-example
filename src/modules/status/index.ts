export * from './dto';
export * from './status.controller';
export * from './status.module';
export * from './status.service';
