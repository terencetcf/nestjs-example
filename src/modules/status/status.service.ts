import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { IConfigs } from '@tt/core/configs';
import { name, version } from '@tt/package';

import { VersionDto } from './dto';

@Injectable()
export class StatusService {
  constructor(private configService: ConfigService<IConfigs>) {}

  getStatus(): VersionDto {
    return {
      env: `${this.configService.get('env')}`,
      name: `${name}`,
      version: `${version}`,
    };
  }
}
