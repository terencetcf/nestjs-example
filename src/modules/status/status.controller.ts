import { Controller, Get } from '@nestjs/common';
import { ApiExcludeEndpoint } from '@nestjs/swagger';

import { VersionDto } from './dto';
import { StatusService } from './status.service';

@Controller('status')
export class StatusController {
  constructor(private readonly statusService: StatusService) {}

  @ApiExcludeEndpoint()
  @Get()
  getStatus(): VersionDto {
    return this.statusService.getStatus();
  }
}
