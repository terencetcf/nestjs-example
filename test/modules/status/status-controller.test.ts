import { HttpStatus } from '@nestjs/common';
import { StatusController, StatusService } from '@tt/modules/status';
import { TestApp } from '@tt/test/setup';

describe('StatusController', () => {
  let app: TestApp;
  beforeAll(async () => {
    app = await TestApp.create().setup([StatusController], [StatusService]);
  });

  afterAll(async () => {
    await app.close();
  });

  describe('GET /status', () => {
    it('should return OK status with expected response body', async () => {
      // Act
      const actual = await app.httpRequest.get(`/status`);

      // Assert
      expect(actual.status).toEqual(HttpStatus.OK);
      expect(actual.body).toMatchSnapshot();
    });
  });
});
