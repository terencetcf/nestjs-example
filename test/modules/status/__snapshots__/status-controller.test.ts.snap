// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusController GET /status should return OK status with expected response body 1`] = `
Object {
  "data": Object {
    "env": "development",
    "name": "nestjs-example",
    "version": "1.0.0",
  },
}
`;
